cpp/src/arrow/dataset/file_rados_parquet.h

```diff
@@ -114,6 +124,7 @@ class ARROW_DS_EXPORT DirectObjectAccess {
     return oid;
   }
 
+  /// \brief Executes query on the librados node.
```

Comment:
This method is quite important. Can we explain it a bit by mentioning the librados::exec call and what it does ?

Reply:
Another thing that occurred to me is for every method you documented, we can add the `/param` options to document and input arguments and also the return type if possible.

Reply:
I did not write the param docs because I saw the IPC and CSV file formats and they didn't have those. But I later found that parquet has them. 
Done.